apps/explorer/src/components/transactions/TransactionsForAddress.tsx

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { useQuery } from '@tanstack/react-query';
+
+import { genTableDataFromTxData } from './TxCardUtils';
+
+import { Banner } from '~/ui/Banner';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { TableCard } from '~/ui/TableCard';
+
+interface Props {
+    address: string;
+    type: 'object' | 'address';
+}
+
+export function TransactionsForAddress({ address, type }: Props) {
+    const rpc = useRpcClient();
+
+    const { data, isLoading, isError } = useQuery(
+        ['transactions-for-address', address, type],
+        async () => {
+            const filters =
+                type === 'object'
+                    ? [{ InputObject: address }, { ChangedObject: address }]
+                    : [{ ToAddress: address }, { FromAddress: address }];
+
+            const results = await Promise.all(
+                filters.map((filter) =>
+                    rpc.queryTransactions({
+                        filter,
+                        order: 'descending',
+                        limit: 100,
+                        options: {
+                            showEffects: true,
+                            showBalanceChanges: true,
+                            showInput: true,
+                        },
+                    })
+                )
+            );
+
+            return [...results[0].data, ...results[1].data].sort(
```

Comment:
this might be a pre-mature optimization, but I recently learned about the `select` option in react-query for doing data transformations which could be nice to use here

https://tkdodo.eu/blog/react-query-data-transformations

Reply:
That makes sense when we want to have a view over the data that we're fetching, but in this case this is defining the actual return data (basically simulating a single list). Eventually this will move to a single `queryTransaction` which will also solve this problem.